opentelemetry-sdk/src/opentelemetry/sdk/_metrics/instrument.py

```diff
@@ -67,9 +72,9 @@ def inner() -> Iterable[Measurement]:
 
             self._callback = inner
 
-    @property
-    def callback(self) -> CallbackT:
-        return self._callback
+    def callback(self):
```

Comment:
Who is gonna call this? In the deisgn, MeasurementConsumer keeps references to all the async instruments so it controls when they get called.

Reply:
Ok, moving it back to a property. Nevertheless, I find it a bit weird that it is the measurement consumer who calls the asynchronous instruments callbacks. Their callbacks are functions analogous to the synchronous instruments `add` or `record` methods who are not being called by the measurement consumer but by the end user. In the case of the asynchronous instruments, it feels like it makes more sense for something external to be the one that calls them, like a sort of end user that calls them periodically and asynchronously.

Reply:
Asynch and synch instruments are inherently different right? Just because an end user makes calls directly to increment sync instruments, doesn't mean the same mechanism has to be used for asynch. In fact, I would think calling a callback externally somehow kind of goes against the design of what a 'callback" is supposed to be?

Reply:
Can you add back the type annotation?
```suggestion
    def callback(self) -> CallbackT:
```